File: construction-legal-lookup-app/src/main/java/com/constructionlegallookup/construction_legal_lookup_app/services/PublicService.java
package com.constructionlegallookup.construction_legal_lookup_app.services;

import java.util.List;

import com.constructionlegallookup.construction_legal_lookup_app.dto.responses.pub.CategoryResponse;
import com.constructionlegallookup.construction_legal_lookup_app.dto.responses.pub.PopularDocumentResponse;
import com.constructionlegallookup.construction_legal_lookup_app.dto.responses.pub.RecentDocumentResponse;
import com.constructionlegallookup.construction_legal_lookup_app.dto.responses.pub.StatsResponse;

public interface PublicService {
    StatsResponse getStats();
    List<RecentDocumentResponse> getRecentDocuments(int limit);
    List<PopularDocumentResponse> getPopularDocuments(int limit);
    List<RecentDocumentResponse> getUpdatedDocuments(int limit);
    List<CategoryResponse> getCategories();
}

File: frontend/src/services/public.service.js
import { api } from './api';

export async function getStats() {
  const { data } = await api.get('/public/stats');
  return data.data;
}

export async function getRecentDocuments(limit = 10) {
  const { data } = await api.get('/public/documents/recent', { params: { limit } });
  return data.data;
}

export async function getPopularDocuments(limit = 10) {
  const { data } = await api.get('/public/documents/popular', { params: { limit } });
  return data.data;
}

export async function getUpdatedDocuments(limit = 10) {
  const { data } = await api.get('/public/documents/updated', { params: { limit } });
  return data.data;
}

export async function getCategories() {
  const { data } = await api.get('/public/categories');
  return data.data;
}
